pkg/gatewayserver/io/io.go

```diff
@@ -688,13 +694,19 @@ func (c *Connection) ScheduleDown(path *ttnpb.DownlinkPath, msg *ttnpb.DownlinkM
 		rx2 = i == 1
 		rxErrs = nil
 		delay = time.Duration(em.Starts() - now)
-		logger.WithFields(log.Fields(
+		logger = logger.WithFields(log.Fields(
 			"rx_window", i+1,
 			"starts", em.Starts(),
 			"duration", em.Duration(),
 			"now", now,
 			"delay", delay,
-		)).Debug("Scheduled downlink")
+			"max_delay", maxDelay,
+		))
+		if delay >= maxDelay {
+			delay = maxDelay
+			logger.Error("Scheduling delay above maximum delay")
+		}
+		logger.Debug("Scheduled downlink")
```

Comment:
The error would reach Sentry, and we can monitor it there. This is at least a start to try to find a pattern of these large delays.

The NS uses the delay in order to estimate when the transmission takes place, in order to avoid scheduling everything at once. Extremely large delay values basically stop downlinks completely, as the NS waits for the downlink to be transmitted before transmitting a new one (and if delay was 1 year, there will be no downlinks for a year).

If we limit the delay, at least we can stop the MAC state from being 'tainted', until we can find out more information about what causes these delays to be so large.